vortex-fastlanes/src/bitpacking/compress.rs

```diff
@@ -101,17 +101,13 @@ impl EncodingCompression for BitPackedEncoding {
     }
 }
 
-fn bitpack(parray: &PrimitiveArray, bit_width: usize) -> ArrayRef {
+fn bitpack(parray: &PrimitiveArray, bit_width: usize) -> VortexResult<ArrayRef> {
     // We know the min is > 0, so it's safe to re-interpret signed integers as unsigned.
     // TODO(ngates): we should implement this using a vortex cast to centralize this hack.
-    let bytes = match parray.ptype() {
-        I8 | U8 => bitpack_primitive(parray.buffer().typed_data::<u8>(), bit_width),
-        I16 | U16 => bitpack_primitive(parray.buffer().typed_data::<u16>(), bit_width),
-        I32 | U32 => bitpack_primitive(parray.buffer().typed_data::<u32>(), bit_width),
-        I64 | U64 => bitpack_primitive(parray.buffer().typed_data::<u64>(), bit_width),
-        _ => panic!("Unsupported ptype {}", parray.ptype()),
-    };
-    PrimitiveArray::from(bytes).into_array()
+    let bytes = match_integers_by_width!(parray.ptype(), |$P| {
```

Comment:
I was looking at this code earlier today... might be easier to reinterpret cast using the function I just added in #189 and then just match on unsigned? But we'd have to add ptype.to_signed/to_unsigned